src/model/guild/guild_id.rs

```diff
@@ -1005,6 +1005,21 @@ impl GuildId {
         http.as_ref().kick_member(self, user_id.into(), Some(reason)).await
     }
 
+    /// Returns a guild [Member] object for the current user.
+    ///
+    /// See [`Http::get_current_user_guild_member`][impl] for more.
+    ///
+    /// [impl]: crate::http::Http::get_current_user_guild_member
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error::Http`] if the current user is not in the guild or the access token lacks
+    /// the necessary scope.
```

Comment:
```suggestion
    /// Returns a guild [`Member`] object for the current user.
    ///
    /// See [`Http::get_current_user_guild_member`] for more.
    ///
    /// [`Http::get_current_user_guild_member`]: crate::http::Http::get_current_user_guild_member
    ///
    /// # Errors
    ///
    /// Returns an [`Error::Http`] if the current user is not in the guild or the access token
    /// lacks the necessary scope.
```
formatting

Also, you should add wrapper methods to `Guild` and `PartialGuild` that call this method on their `id` field.

Reply:
I've pulled in the formatting suggestions and added the wrapper methods to `Guild` and `PartialGuild`.